ironfish/src/account/accounts.ts

```diff
@@ -374,39 +376,21 @@ export class Accounts {
       }
 
       for (const account of accounts) {
-        // Try decrypting the note as the owner
-        const receivedNote = note.decryptNoteForOwner(account.incomingViewKey)
-        if (receivedNote) {
-          if (receivedNote.value() !== BigInt(0)) {
-            notes.push({
-              noteIndex: currentNoteIndex,
-              forSpender: false,
-              merkleHash: note.merkleHash().toString('hex'),
-              nullifier:
-                currentNoteIndex !== null
-                  ? Buffer.from(
-                      receivedNote.nullifier(account.spendingKey, BigInt(currentNoteIndex)),
-                    ).toString('hex')
-                  : null,
-              account: account,
-            })
-          }
-          continue
-        }
-
-        // Try decrypting the note as the spender
-        const spentNote = note.decryptNoteForSpender(account.outgoingViewKey)
-        if (spentNote) {
-          if (spentNote.value() !== BigInt(0)) {
-            notes.push({
-              noteIndex: currentNoteIndex,
-              forSpender: true,
-              merkleHash: note.merkleHash().toString('hex'),
-              nullifier: null,
-              account: account,
-            })
-          }
-          continue
+        const decryptedNote = await this.workerPool.decryptNotes(
+          note.serialize(),
+          account.incomingViewKey,
+          account.outgoingViewKey,
+          account.spendingKey,
+          currentNoteIndex,
+        )
```

Comment:
I think this code is REALLY clean and easy to read.

```suggestion
        )

```